Introduce dynamic table append (#15897)
apache/druid: docs/querying/datasource.md

Refer to the [Query execution](query-execution.md#union) page for more details on how queries are executed when you
use union datasources.

> ### `dynamic table append`

Reviewer: Since this isn't a property value, we don't need code font.
```suggestion
### Dynamic table append
```

### `inline`

<Tabs>

> <TabItem value="sql" label="SQL">

Reviewer: Thanks for adding this missing example

Refer to the [Query execution](query-execution.md#inline) page for more details on how queries are executed when you
use inline datasources.

> ### Dynamic table append `TABLE(APPEND(...))`

Reviewer: * Since this has the same native query syntax as `union` should this content go within the existing `union` section?
* Suggest removing the SQL syntax from the header title, so just have `#### Dynamic table append`

Reviewer: I like both of those ideas. Since the native union behavior is the same, the new content can go in there, and removing the repeated `native` section won't have people wondering what the difference is.

> ### Dynamic table append `TABLE(APPEND(...))`

Reviewer: Consider also including a line / link to this content from the [SQL syntax doc](https://druid.apache.org/docs/latest/querying/sql#union-all) in https://github.com/apache/druid/blob/master/docs/querying/sql.md?plain=1#L322

Author: referenced updated the query and referenced [table-level union](sql.md#table-level) as that's more closer to what the user might be working with
thank you!

Author: I've added a backreferencing paragraph to that as well

> This is essentially an SQL level syntactic sugar which matches columns by name from multiple tables.

Reviewer: ```suggestion
Perform dynamic table appends in SQL using `TABLE(APPEND(...))`. This simplifies SQL syntax to match columns by name from multiple tables. The native query syntax remains the same as for native union datasources.
```

> Suppose you have 3 tables:

Reviewer: ```suggestion
Suppose you have three tables:
```

* `table1` has `column1`
* `table2` has `column2`
* `table3` has `column1`, `column2`, `column3`

> It is possible to create an union view of the above tables is possible by using UNION ALL:

Reviewer: ```suggestion
You can create a union view of all the tables by using UNION ALL:
```

```sql
SELECT column1,NULL AS column2,NULL AS column3 FROM table1
UNION ALL
SELECT NULL AS column1,column2,NULL AS column3 FROM table2
UNION ALL
SELECT column1,column2,column3 FROM table3
```

However depending on the size of the table's schema it might be quite complicated to do that; `TABLE(APPEND('table1','table2','table3'))` represents the same in a more compact form.

> <Tabs>

Reviewer: All the sections that have these tabs do so at the beginning of the section, so consider moving this to the top for consistency
